Sources/AuthFoundation/OAuth2/OAuth2Client.swift

```diff
@@ -424,31 +424,57 @@ public final class OAuth2Client {
         
         // Exchange the token
         request.send(to: self) { result in
-            guard case let .success(response) = result else {
-                completion(result)
-                return
-            }
-            
-            // Perform idToken/accessToken validation
-            do {
-                try response.result.validate(using: self, with: request as? IDTokenValidatorContext)
-            } catch {
-                completion(.failure(.validation(error: error)))
-                return
-            }
-            
             // Wait for the JWKS keys, if necessary
-            group.notify(queue: DispatchQueue.global()) {
+            group.notify(queue: DispatchQueue.global()) { [weak self] in
+                // Perform idToken/accessToken validation
+                self?.validateToken(
+                    request: request,
+                    keySet: keySet,
+                    oauthTokenResponse: result,
+                    completion: completion
+                )
+            }
+        }
+    }
+    
+    private func validateToken<T: OAuth2TokenRequest>(
+        request: T,
+        keySet: JWKS?,
+        oauthTokenResponse: Result<APIResponse<Token>, APIClientError>,
+        completion: @escaping (Result<APIResponse<Token>, APIClientError>) -> Void
+    ) {
+        guard case let .success(response) = oauthTokenResponse else {
+            completion(oauthTokenResponse)
+            return
+        }
+        
+        // Retrieves the org's OpenID configuration
+        self.openIdConfiguration { result in
+            switch result {
+            case .failure(let error):
+                completion(.failure(.serverError(error)))
+            case .success(let configuration):
+                do {
+                    try response.result.validate(
```

Comment:
Similar thing here. Elsewhere in the project, this is the formatting style used, so I just want to keep it consistent.